pennylane/_device.py

```diff
@@ -25,9 +25,22 @@
 import numpy as np
 
 import pennylane as qml
-from pennylane.measurements import Expectation, MidMeasure, Probability, Sample, State, Variance
-from pennylane.operation import Observable, Operation, Tensor
-from pennylane.wires import WireError, Wires
+from pennylane.operation import (
+    Operation,
+    Observable,
+    Tensor,
+)
+from pennylane.measurements import (
+    Sample,
+    State,
+    Variance,
+    Expectation,
+    Probability,
+    MidMeasure,
+    ShadowExpval,
+)
```

Comment:
Definitely this is not ideal for users and developers (to need to modify the device every time a new measurement is added!)

Something to think about for the future: can we add an API for users/developers to define _new measurement processes_ that contain the classical post-processing, and can be used by default by any device?

Reply:
💯

Reply:
Definitely something we can think about in the next release 🙂